src/Components/DateSearch.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from "react";
+import TextField from "@mui/material/TextField";
+import Stack from "@mui/material/Stack";
+import Autocomplete from "@mui/material/Autocomplete";
+import { fetchData } from "../Utils/FetchData";
+
+const apiUrl = "https://6603f6ac2393662c31d04103.mockapi.io/gatewayclients/api/gateways";
```

Comment:
Avoid hardcoding API links. Consider retrieving them from environment files. Refer to the link below for more information: https://create-react-app.dev/docs/adding-custom-environment-variables/